CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/io/DraggableCellNGTest.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2010-2021 Australian Signals Directorate
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.dataaccess.io;
+
+import java.awt.GraphicsEnvironment;
+import java.util.HashSet;
+import java.util.Set;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.control.ListView;
+import javafx.scene.input.DragEvent;
+import javafx.scene.input.Dragboard;
+import javafx.scene.input.TransferMode;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author antares
+ */
+public class DraggableCellNGTest {
+    
+    public DraggableCellNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of updateItem method, of class DraggableCell with null item.
+     */
+    @Test
+    public void testUpdateItemNullItem() {
+        System.out.println("updateItemNullItem");
+        
+        // TODO: Find a way to instantiate toolkit in a headless environment
+        // This unit test should pass locally but will hold up in a headless environment (e.g. CI)
+        // Putting it in this if loop is a temp fix for the hold up in CI but it does it by skipping the test 
+        // (so CI won't actually tell you whether this is passing or not)
+        if (!GraphicsEnvironment.isHeadless()) {
+            new JFXPanel();
+            
+            final DraggableCell<String> instance = new DraggableCell<>();
+            instance.updateItem(null, false);
+
+            assertEquals(instance.getText(), null);
+            assertEquals(instance.getItem(), null);
+            assertEquals(instance.isEmpty(), false);
```

Comment:
Why is it that when you pass it null, its not empty, and when you pass it an object it returns true to the call is.Empty() ?

Reply:
@aldebaran30701 DraggableCell (or rather its super class Cell) has a property called `empty` so `isEmpty()` is actually checking whether that property value is false, not whether the cell contains any content (admittedly confusing given the context of `isEmpty()` used for just about any other Java object). So the 2nd parameter in `updateItem()` is setting the empty property so that's the value that is determining my assert